bagitobjecttransfer/recordtransfer/views.py

```diff
@@ -15,100 +18,129 @@
+    send_thank_you_for_your_transfer,
+    send_user_activation_email,
+    send_your_transfer_did_not_go_through,
+)
 from recordtransfer.forms import SignUpForm, UserProfileForm
+from recordtransfer.models import (
+    SavedTransfer,
+    Submission,
+    SubmissionGroup,
+    UploadedFile,
+    UploadSession,
+    User,
+)
 from recordtransfer.tokens import account_activation_token
+from recordtransfer.utils import get_human_readable_file_count, get_human_readable_size
 
-
-LOGGER = logging.getLogger('recordtransfer')
+LOGGER = logging.getLogger("recordtransfer")
 
 
 class Index(TemplateView):
-    ''' The homepage '''
-    template_name = 'recordtransfer/home.html'
+    """The homepage"""
+
+    template_name = "recordtransfer/home.html"
 
 
 class TransferSent(TemplateView):
-    ''' The page a user sees when they finish a transfer '''
-    template_name = 'recordtransfer/transfersent.html'
+    """The page a user sees when they finish a transfer"""
+
+    template_name = "recordtransfer/transfersent.html"
 
 
 class SystemErrorPage(TemplateView):
-    """ The page a user sees when there is some system error. """
-    template_name = 'recordtransfer/systemerror.html'
+    """The page a user sees when there is some system error."""
+
+    template_name = "recordtransfer/systemerror.html"
 
 
 class UserProfile(UpdateView):
-    ''' This view shows two things:
+    """View to show two things:
     - The user's profile information
-    - A list of the Submissions a user has created via transfer
-    '''
+    - A list of the Submissions a user has created via transfer.
+    """
 
-    template_name = 'recordtransfer/profile.html'
+    template_name = "recordtransfer/profile.html"
     paginate_by = 10
     form_class = UserProfileForm
-    success_url = reverse_lazy('recordtransfer:userprofile')
+    success_url = reverse_lazy("recordtransfer:userprofile")
 
     def get_object(self, queryset=None):
         return self.request.user
 
     def get_context_data(self, **kwargs):
         context = super(UserProfile, self).get_context_data(**kwargs)
-        context['in_process_submissions'] = SavedTransfer.objects.filter(user=self.request.user)\
-            .order_by('-last_updated')
-        context['user_submissions'] = Submission.objects.filter(user=self.request.user).order_by('-submission_date')
+        context["in_process_submissions"] = SavedTransfer.objects.filter(
+            user=self.request.user
+        ).order_by("-last_updated")
+        context["user_submissions"] = Submission.objects.filter(user=self.request.user).order_by(
+            "-submission_date"
+        )
         return context
 
     def form_valid(self, form):
-        messages.success(self.request, 'Preferences updated')
-        return super().form_valid(form)
+        response = super().form_valid(form)
+        update_session_auth_hash(self.request, form.instance)
+        messages.success(self.request, "Preferences updated")
+        return response
```

Comment:
Three things here:

1. If the password was changed, "Preferences updated" should read "Password updated successfully"
2. `update_session_auth_hash` should only be called if the password was changed
3. Can you send an email like in `recordtransfer.admin.CustomUserAdmin.user_change_password` when the user's password is changed? The same email function can be used.

Reply:
1) If the user also changed their email notification preference, what message should I display?

Reply:
I would say the password message can override the email notification preference. So if they both change, just show the password changed in the email

Reply:
I added that functionality here: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/208/commits/abcb88874eeb72b269501482fce394db2c9c236f

However, I'm not sure how `.delay()` is being used, since I couldn't find any reference to it. Looking at other places where it's used, I see the same thing, such as in `views.CreateAccount`:

![image](https://github.com/user-attachments/assets/4010c457-5182-42af-bb4e-cfddb3b49d52)

Reply:
Also, is there an actual email server set up to send emails for testing purposes?

Reply:
Yes, in the development application you can go to localhost:8025 to access any emails - any email sent is intercepted by MailHog so you can view them there

Reply:
> However, I'm not sure how `.delay()` is being used, since I couldn't find any reference to it. Looking at other places where it's used, I see the same thing, such as in `views.CreateAccount`:

See docs here for how `delay()` works. https://github.com/rq/django-rq?tab=readme-ov-file#job-decorator